src/Swarm/TUI/Controller.hs

```diff
@@ -808,27 +816,30 @@ adjustTPS (+/-) = uiState . lgTicksPerSecond %~ (+/- 1)
 ------------------------------------------------------------
 
 -- | Handle user input events in the robot panel.
-handleRobotPanelEvent :: AppState -> BrickEvent Name AppEvent -> EventM Name (Next AppState)
-handleRobotPanelEvent s (Key V.KEnter) =
-  maybe (continueWithoutRedraw s) (descriptionModal s) (focusedEntity s)
-handleRobotPanelEvent s (CharKey 'm') =
-  maybe (continueWithoutRedraw s) (makeEntity s) (focusedEntity s)
-handleRobotPanelEvent s (CharKey '0') = do
-  continue $ s & (uiState . uiShowZero %~ not) . (uiState . uiInventoryShouldUpdate .~ True)
-handleRobotPanelEvent s (VtyEvent ev) = do
-  let mList = s ^? uiState . uiInventory . _Just . _2
+handleRobotPanelEvent :: BrickEvent Name AppEvent -> EventM Name AppState ()
+handleRobotPanelEvent (Key V.KEnter) = do
+  focused <- gets focusedEntity
+  maybe continueWithoutRedraw descriptionModal focused
```

Comment:
Would `gets focusedEntity >>= maybe continueWithoutRedraw descriptionModal` be simpler?

Reply:
Sure, I almost want a name for a "gets maybe or no redraw" helper function, but that can be done later. :+1: